lib/engine/game/g_18_india/step/railhead_tracker.rb

```diff
@@ -0,0 +1,85 @@
+# frozen_string_literal: true
+
+require_relative '../../../step/tracker'
+
+module Engine
+  module Game
+    module G18India
+      module Step
+        module RailheadTracker
+          # Implements track laying rule for multiple yellow tiles
+          # Track laying starts at a railhead(token) and must extend a route(path) from all laid tiles
+
+          def setup
+            super
+            @round.next_empty_hexes = nil
+            @round.laid_yellow_hexes = []
+          end
+
+          def round_state
+            super.merge({ next_empty_hexes: nil, laid_yellow_hexes: [] })
+          end
+
+          def calculate_railhead_hexes
+            return [] if @round.laid_yellow_hexes.empty?
+
+            # check simple case of only one or two 'white' neighbor connected to prior tile => return without walking
+            last_tile = @round.laid_yellow_hexes.last.tile
+            unless [1, 6].include?(last_tile.exits.size) # exclude triple town tiles (6 exits) and OO tiles (1 exit)
+              empty_neighbors = empty_neighbors(last_tile.hex, last_tile.exits)
+              LOGGER.debug "railhead_hexes >> empty_neighbors: #{empty_neighbors.inspect}"
+              return empty_neighbors if [1, 2].include?(empty_neighbors.size)
+            end
+
+            corp = @round.current_operator
+            railheads = corp.placed_tokens.map(&:city)
+            placed_paths = @round.laid_yellow_hexes.map(&:tile).map(&:paths) # paths on all previously laid yellow hexes
+            LOGGER.debug "railhead_hexes >> corp: #{corp.inspect}, railheads: #{railheads.inspect}"
+            return nil if railheads.empty? # Can happen if OO token is waiting to be placed
+
+            next_hexes = []
+            railheads.each do |railhead|
+              railhead.walk(corporation: corp) do |path, visited_paths, _visited|
+                # check if path ends at targeted "white" hex
+                empty_neighbors = empty_neighbors(path.hex, path.exits)
+                next if empty_neighbors.empty?
+
+                # check if visited path has at least one path from all placed tiles (triple town tiles may have unused paths)
+                placed_tiles_visited = placed_paths.map { |paths_on_tile| !(paths_on_tile & visited_paths.keys).empty? }
+                next unless placed_tiles_visited.all?
+
+                # confirm that placed tiles are visited in the correct sequence along the visited path
+                sequence = placed_paths.map { |tile_paths| tile_paths.map { |p| visited_paths.keys.index(p) }.compact.min }
+                next unless sequence.each_cons(2).all? { |x, y| x < y }
+
+                LOGGER.debug " >> path found to hex: #{empty_neighbors.inspect}, sequence: #{sequence.inspect}"
+                next_hexes.concat(empty_neighbors)
```

Comment:
Do you need an additional check that the route from the last placed yellow tile to these white hexes only passes through grey or red tiles?

Reply:
It is OK if the route passes through other tiles of any color that are pre-printed or previously placed.  I have confirmed that with the designer.  

> > Can the track laying continue past a tile placed on a prior turn (yellow / green / brown / gray) and continue laying past that connection? 
**yes**